pandas/core/frame.py

```diff
@@ -7213,38 +7213,87 @@ def _get_agg_axis(self, axis_num):
 
     def mode(self, axis=0, numeric_only=False, dropna=True):
         """
-        Gets the mode(s) of each element along the axis selected. Adds a row
-        for each mode per label, fills in gaps with nan.
+        Get the mode(s) of each element along the selected axis.
 
-        Note that there could be multiple values returned for the selected
-        axis (when more than one item share the maximum frequency), which is
-        the reason why a dataframe is returned. If you want to impute missing
-        values with the mode in a dataframe ``df``, you can just do this:
-        ``df.fillna(df.mode().iloc[0])``
+        The mode of a set of values is the value that appears most often.
+        It can be multiple values.
 
         Parameters
         ----------
         axis : {0 or 'index', 1 or 'columns'}, default 0
-            * 0 or 'index' : get mode of each column
-            * 1 or 'columns' : get mode of each row
-        numeric_only : boolean, default False
-            if True, only apply to numeric columns
-        dropna : boolean, default True
+            The axis to iterate over while searching for the mode.
+            To find the mode for each column, use ``axis='index'``.
+            To find the mode for each row, use ``axis='columns'``.
```

Comment:
I like the first line of your summary.  However, I would put back the two bullet points.

Reply:
Agree, fixed. And we should probably define a standard and be consistent when documenting `axis` (added it to #20298)